src/components/blipDatepicker/index.js

```diff
@@ -0,0 +1,615 @@
+      this._createTimeStructure()
+    }
+
+    this._addEventListeners()
+
+    this.monthDate = new Date(monthDate)
+
+    return this._datepicker
+  }
+
+  _createMonthTable() {
+    var monthTable = this._createHTMLElement('table', BlipDatepicker.style.monthTable)
+    var tableHead = monthTable.createTHead()
+    var tableBody = monthTable.createTBody()
+
+    var headerRow = tableHead.insertRow()
+
+    var titleCell = this._createHTMLElement('th')
+    var prevCell = this._createHTMLElement('th')
+    var nextCell = this._createHTMLElement('th')
+
+    var monthTitle = this._createHTMLElement('span', BlipDatepicker.style.monthTitle)
+    var yearInput = this._createHTMLElement('input', BlipDatepicker.style.yearInput)
+    yearInput.type = 'number'
+
+    var createButton = (icon, value, ...classes) => {
+      var button = html`<button class="${BlipDatepicker.style.monthButton}">${raw(icon)}</button>`
+      if (classes) classes.forEach(className => button.classList.add(className))
+      button.value = value
+      return button
+    }
+
+    var prevButton = createButton(ArrowLeft, '-1', BlipDatepicker.style.monthPrev)
+    var nextButton = createButton(ArrowRight, '1', BlipDatepicker.style.monthNext)
+
+    titleCell.colSpan = '5'
+    titleCell.appendChild(monthTitle)
+    titleCell.appendChild(yearInput)
+    prevCell.appendChild(prevButton)
+    nextCell.appendChild(nextButton)
+
+    headerRow.appendChild(prevCell)
+    headerRow.appendChild(titleCell)
+    headerRow.appendChild(nextCell)
+
+    var weekdayRow = tableBody.insertRow()
+    weekdayRow.classList.add(BlipDatepicker.style.weekdayWeek)
+    for (var day = 0; day < BlipDatepicker.weekSize; day++) {
+      var weekday = weekdayRow.insertCell()
+      weekday.classList.add(BlipDatepicker.style.weekDay)
+      weekday.innerText = this.i18n.weekdays[day][0]
+      weekday.title = this.i18n.weekdays[day]
+    }
+
+    var daysRows = []
+    var monthDays = []
+    for (var week = 0; week < BlipDatepicker.monthRows; week++) {
+      var weekRow = tableBody.insertRow()
+      for (day = 0; day < BlipDatepicker.weekSize; day++) {
+        var monthDay = weekRow.insertCell()
+        monthDay.classList.add(BlipDatepicker.style.monthDay)
+
+        var dayDiv = this._createHTMLElement('div')
+        monthDay.appendChild(dayDiv)
+
+        monthDays.push(monthDay)
+      }
+      daysRows.push(weekRow)
+    }
+
+    this._datepicker.appendChild(monthTable)
+
+    this._monthTable = monthTable
+    this._monthTitle = monthTitle
+    this._yearInput = yearInput
+    this._prevButton = prevButton
+    this._nextButton = nextButton
+    this._tableBody = tableBody
+    this._daysRows = daysRows
+    this._monthDays = monthDays
+  }
+
+  _createDateSelector() {
+    var dateSelector = this._createHTMLElement('div', BlipDatepicker.style.dateSelector)
+
+    var selectorOptions = []
+    var selectorInputs = []
+    for (var row = 0; row < BlipDatepicker.selectorRows; row++) {
```

Comment:
`row` variable are not be used by this context and `option` either. Won't be better to use `map`, `reduce` or even `forEach` methods?

Reply:
The selector has an arbitrary number of rows and columns, the `for` was the best way to do it